---
title: "ILRUN_PNAS"
author: "Marina Alexander"
date: "28/08/2020"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(dplyr)
library(readr)
library(vroom)
library(stringr)
library(EnhancedVolcano)
library(viridis)
library(ggsci)
library(cowplot)
library(patchwork)
library(ggsignif)


knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```



```{r Figure 1A siRNA validation  }

sirna_valid <- read.csv("data/siRNA_validation.csv")

###perform ANOVA ########
siRNA_aov <- aov(Ct ~ siRNA, data = sirna_valid)
summary(siRNA_aov)

###Summarise for plotting bar graph #####
siRNA_valid_stats <- sirna_valid %>% 
  group_by(siRNA) %>% 
  summarise(mean = mean(Ct), sd = sd(Ct)) %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd)

#####order the x axis non-alphabetical##############
siRNA_valid_stats$siRNA <- factor(siRNA_valid_stats$siRNA,levels =c("siNEG", "siILRUN"))

##### plot the data ################
Figure1A <- ggplot(siRNA_valid_stats, aes(y= mean, x=siRNA))+
  geom_bar(stat= "Identity", color= "black", fill = "grey", position = position_dodge()) +
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  geom_signif(comparisons=list(c("siNEG", "siILRUN")), annotations="***", y_position = 1.1, tip_length = 0, vjust=0.4) +
  scale_y_continuous(breaks=c(0, 0.25,0.5, 0.75, 1.0))+
  theme_set(theme_classic(base_size = 8)) +
  theme(axis.title.x = element_text(size = 8))+
  labs(tag = "A",
       y ="Relative ILRUN mRNA",
       x = "")
  
ggsave(
  filename = "results/Figure1A.TIFF",
  plot = Figure1A,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 3.4,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)



```

```{r Figure 1B polyI:C }

##### performing ANOVA on experimental groups ################

polyIC_reps_IFNb <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(treatment == "poly(I:C)") %>% 
  filter(target == "IFNb")

polyIC_reps_TNF <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(treatment == "poly(I:C)") %>% 
  filter(target == "TNFa")

polyIC_reps_IL6 <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(treatment == "poly(I:C)") %>% 
  filter(target == "IL6")
####################################################
polyIC_reps_IFNb_treat_NEG <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siNEG") %>% 
  filter(target == "IFNb")

polyIC_reps_TNF_treat_NEG <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siNEG") %>% 
  filter(target == "TNFa")

polyIC_reps_IL6_treat_NEG <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siNEG") %>% 
  filter(target == "IL6")
####################################################
polyIC_reps_IFNb_treat_ILRUN <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siILRUN") %>% 
  filter(target == "IFNb")

polyIC_reps_TNF_treat_ILRUN <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siILRUN") %>% 
  filter(target == "TNFa")

polyIC_reps_IL6_treat_ILRUN <- read.csv("data/Caco2_polyIC_reps.csv") %>%
  filter(siRNA == "siILRUN") %>% 
  filter(target == "IL6")
##################################################
polyIC_aov_IFNb <- aov(relative_mRNA ~ siRNA, data = polyIC_reps_IFNb)
summary(polyIC_aov_IFNb)

polyIC_aov_TNF <- aov(relative_mRNA ~ siRNA, data = polyIC_reps_TNFa)
summary(polyIC_aov_TNFa)

polyIC_aov_IL6 <- aov(relative_mRNA ~ siRNA, data = polyIC_reps_IL6)
summary(polyIC_aov_IL6)
##################################################
polyIC_aov_IFNb_treat_NEG <- aov(relative_mRNA ~ treatment, data = polyIC_reps_IFNb_treat_NEG)
summary(polyIC_aov_IFNb_treat_NEG)

polyIC_aov_TNF_treat_NEG <- aov(relative_mRNA ~ treatment, data = polyIC_reps_TNF_treat_NEG)
summary(polyIC_aov_TNF_treat_NEG)

polyIC_aov_IL6_treat_NEG <- aov(relative_mRNA ~ treatment, data = polyIC_reps_IL6_treat_NEG)
summary(polyIC_aov_IL6_treat_NEG)
##################################################
polyIC_aov_IFNb_treat_ILRUN <- aov(relative_mRNA ~ treatment, data = polyIC_reps_IFNb_treat_ILRUN)
summary(polyIC_aov_IFNb_treat_ILRUN)

polyIC_aov_TNF_treat_ILRUN <- aov(relative_mRNA ~ treatment, data = polyIC_reps_TNF_treat_ILRUN)
summary(polyIC_aov_TNF_treat_ILRUN)

polyIC_aov_IL6_treat_ILRUN <- aov(relative_mRNA ~ treatment, data = polyIC_reps_IL6_treat_ILRUN)
summary(polyIC_aov_IL6_treat_ILRUN)


##### data of means  ################
polyIC <- read.csv("data/Caco2_polyIC.csv") %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd)

#####order the x axis non-alphabetical##############
polyIC$siRNA <- factor(polyIC$siRNA,levels =c("siNEG", "siILRUN"))
polyIC$treatment <- factor(polyIC$treatment,levels =c("control", "poly(I:C)"))

##### plot the data ################
 IFNb <- polyIC %>% 
  filter(target == "IFNb") %>% 
  ggplot(aes(y= mean, x=siRNA, fill = treatment))+
  geom_bar(stat= "Identity", color= "black", position = position_dodge()) +
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  geom_signif(comparisons=list(c("siNEG", "siILRUN")), annotations="**", y_position = 1600, tip_length = 0, vjust=0.4) +
  geom_signif(y_position=c(500, 1400), xmin=c(0.8, 1.8), xmax=c(1.2, 2.2),
              annotation=c("**", "**"), tip_length=0) +
  scale_fill_manual(values = c("#56B4E9", "#0072B2"))+
  theme_set(theme_classic(base_size = 8)) +
  theme(legend.key.size = unit(0.3, "cm")) +
  labs(title = "IFNb",
       y ="Relative mRNA",
       x = "")+
  theme(plot.title = element_text(hjust = 0.5, face = "italic"))

 TNF <- polyIC %>% 
  filter(target == "TNFa") %>% 
  ggplot(aes(y= mean, x=siRNA, fill = treatment))+
  geom_bar(stat= "Identity", color= "black", position = position_dodge()) +
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
   geom_signif(comparisons=list(c("siNEG", "siILRUN")), annotations="*", y_position = 36, tip_length = 0, vjust=0.4) +
   geom_signif(y_position=c(12, 32), xmin=c(0.8, 1.8), xmax=c(1.2, 2.2),
              annotation=c("*", "**"), tip_length=0) +
  scale_fill_manual(values = c("#56B4E9", "#0072B2"))+
  theme_set(theme_classic(base_size = 8)) +
  labs(title = "TNF",
       y ="",
       x = "")+
  theme(plot.title = element_text(hjust = 0.5, face = "italic"))
 
  IL6 <- polyIC %>% 
  filter(target == "IL6") %>% 
  ggplot(aes(y= mean, x=siRNA, fill = treatment))+
  geom_bar(stat= "Identity", color= "black", position = position_dodge()) +
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
    geom_signif(comparisons=list(c("siNEG", "siILRUN")), annotations="**", y_position = 32, tip_length = 0, vjust=0.4) +
    geom_signif(y_position=c(16, 28), xmin=c(0.8, 1.8), xmax=c(1.2, 2.2),
              annotation=c("**", "**"), tip_length=0) +
  scale_fill_manual(values = c("#56B4E9", "#0072B2"))+
  theme_set(theme_classic(base_size = 8)) +
  labs(title = "IL6",
       y ="",
       x = "")+
  theme(plot.title = element_text(hjust = 0.5, face = "italic"))
  
legend <- get_legend(IFNb) 

plot_cytokines <- plot_grid(IFNb + theme(legend.position="none"),
                  TNF + theme(legend.position="none"),
                  IL6 + theme(legend.position="none"), nrow =1)

Figure1B <- plot_grid(plot_cytokines, legend, rel_widths = c(3, .3))
  
ggsave(
  filename = "results/Figure1B.TIFF",
  plot = Figure1B,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 14.6,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)



```

```{r Figure 1C kinetics titres  }

kinetics_titres <- read.csv("data/kinetics_titres.csv")


###Summarise for plotting bar graph #####
kinetics_titres_stats <- kinetics_titres %>% 
  group_by(timepoint) %>% 
  summarise(mean = mean(TCID50), sd = sd(TCID50)) %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd) %>%
  mutate(timepoint = as.numeric(timepoint))

##### plot the data ################
Figure1C <- ggplot(kinetics_titres_stats, aes(y= mean, x=timepoint))+
  geom_line()+
  geom_point()+
  ylim(0,100000)+
  scale_x_continuous(breaks=c(0,3, 6, 12, 24, 48))+
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  theme_set(theme_classic(base_size = 8)) +
  labs(x = "post-infection (h)",
       y = expression(paste(log[10],TCID[50], mL^-1)),
       tag = "C")


ggsave(
  filename = "results/Figure1C.TIFF",
  plot = Figure1C,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 6,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)

```

```{r Figure 1D kinetics RNA  }

kinetics_RNA <- read.csv("data/kinetics_RNA.csv")

###Summarise for plotting bar graph #####
kinetics_RNA_stats <- kinetics_RNA %>% 
  group_by(timepoint) %>% 
  summarise(mean = mean(Ct), sd = sd(Ct)) %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd) %>%
  mutate(timepoint = as.numeric(timepoint))

##### plot the data ################
Figure1D <- ggplot(kinetics_RNA_stats, aes(y= mean, x=timepoint))+
  geom_line()+
  geom_point()+
  ylim(0, 10)+
  scale_x_continuous(breaks=c(0,3, 6, 12, 24, 48))+
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  theme_set(theme_classic(base_size = 8)) +
  labs(y = "Relative SARS-CoV-2 RNA \n (fold change over inoculum)",
       x = "post-infection (h)",
       tag = "D")
  
ggsave(
  filename = "results/Figure1D.TIFF",
  plot = Figure1D,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 6,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)

```


```{r Figure 1E kinetics protein  }

kinetics_protein <- read.csv("data/kinetics_protein.csv")


###Summarise for plotting bar graph #####
kinetics_protein_stats <- kinetics_protein %>% 
  group_by(timepoint) %>% 
  summarise(mean = mean(protein), sd = sd(protein)) %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd) %>%
  mutate(timepoint = as.numeric(timepoint))

##### plot the data ################
Figure1E <- ggplot(kinetics_protein_stats, aes(y= mean, x=timepoint))+
  geom_line()+
  geom_point()+
  ylim(0, 150)+
  scale_x_continuous(breaks=c(0,3, 6, 12, 24, 48))+
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  theme_set(theme_classic(base_size = 8)) +
  labs(y = "Relative SARS-CoV-2 N protein \n (fold change over inoculum)",
       x = "post-infection (h)",
       tag = "E")
  
ggsave(
  filename = "results/Figure1E.TIFF",
  plot = Figure1E,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 6,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)

```


```{r Figure 2A. PCA plot }

##### PCA plot ########

#read in the metadata
ILRUNgenesUninf_metadata <- read_csv("data/ilrun_metadata_siRNA_LT07_rev.csv") %>% 
  filter(Infection == "Uninfected") %>% 
  filter(timepoint == "24hr") %>% 
  filter(Lane == "L001") %>%
  mutate(Name = str_extract(Sample, "^...."))

#join with the counts data 
ILRUNgenesUninf_expression <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_all_normalized_counts.csv", header=TRUE) %>%
  gather(Sample, expression, -gene_locus) %>%
  filter(str_detect(Sample, 'L001')) %>% 
  right_join(ILRUNgenesUninf_metadata, by = "Sample") 

#scale gene expression for all samples 
ILRUNgenesUninf_expression_scaled_genes <- ILRUNgenesUninf_expression %>%
  spread(gene_locus, expression) %>%
  select(-Infection, -siRNA, -timepoint, -Condition, -Lane, -HTSeq_rev, -Name) %>% 
  column_to_rownames("Sample") %>% 
  scale()
#use the prcomp function on scaled samples
ILRUNgenesUninf_pca_genes <- prcomp(ILRUNgenesUninf_expression_scaled_genes)
#tidy data frame for plotting
PCA_data_ILRUNgenesUninf_pca_genes <- ILRUNgenesUninf_pca_genes$x %>% 
  as_tibble(rownames = "Sample") %>%
  gather(PC, expression, -Sample) %>% 
  left_join(ILRUNgenesUninf_metadata, by = "Sample") %>%
  spread(PC, expression)

PCA_data_ILRUNgenesUninf_pca_genes$siRNA <- factor(PCA_data_ILRUNgenesUninf_pca_genes$siRNA,levels =c("NEG", "ILRUN"))

#plot the data
Figure2A <- ggplot(PCA_data_ILRUNgenesUninf_pca_genes, aes(x = PC1, y = PC2)) +
  geom_point(aes(fill = siRNA),color = "black", pch = 21, size = 3)+
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  theme_set(theme_classic(base_size = 10)) +
  theme(legend.position="bottom") +
  labs(tag = "A")

ggsave(
  filename = "results/Figure2A.TIFF",
  plot = Figure2A,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 6,
  height = 7,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)


```

```{r Figure 2B. }

######### Volcano plot #################

p1 <- EnhancedVolcano(forplotting_ILRUNgenesUninf24hr,
                                                   lab = rownames(forplotting_ILRUNgenesUninf24hr),
                                                   x = 'log2FoldChange',
                                                   y = 'padj',
                                                   selectLab = c(immune_genes_Uninf24hr$value),
                                                   xlim = c(-3, 3),
                                                   pCutoff = 0.05,
                                                   FCcutoff = 1,
                                                   pointSize = 1.0,
                                                   labSize = 2.0,
                                                   labCol = 'black',
                                                   labFace = 'bold',
                                                   boxedLabels = TRUE,
                                                   colAlpha = 4/5,
                                                   legendPosition = 'right',
                                                   legendLabSize = 14,
                                                   legendIconSize = 4.0,
                                                   drawConnectors = TRUE,
                                                   widthConnectors = 0.3,
                                                   colConnectors = 'black')

 Figure2B <-  p1 +
   ggplot2::theme_set(theme_classic(base_size = 12)) +
   ggplot2::theme(legend.position = "none",
                  plot.title = element_blank(),
                  plot.subtitle = element_blank(),
                  plot.caption = element_blank()) +
   ggplot2::labs(tag = "B")


ggsave(
  filename = "results/Figure2B.TIFF",
  plot = Figure2B,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 12 ,
  height = 8,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)



```

```{r Figure 2C validation of IL6R, VIM and ISG15}

DEgene_validation <- read.csv("data/RTPCR_validation.csv") %>% 
  mutate(upper = log2FoldChange + lfcSE) %>% 
  mutate(lower = log2FoldChange - lfcSE) %>%
  mutate(direction = if_else(log2FoldChange >= 0, 'UP', 'DOWN'))

dir <- c("UP" = "red", "DOWN" = "blue")  

Figure2C <- ggplot(DEgene_validation, aes(y= log2FoldChange, x=GENE_ID, fill = direction))+
  geom_bar(stat= "Identity", color= "black", position = position_dodge()) +
  facet_wrap(~ assay, dir = "h")+
  geom_errorbar(aes(ymin = lower, ymax = upper), width =0.2, position = position_dodge(0.9)) +
  ylim(-5, 7)+
  scale_fill_manual(values = dir)+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.3, "cm"))+
  labs(tag = "C",
       y = expression(paste(Log[2], " fold change")),
       x = "gene")

ggsave(
  filename = "results/Figure2C.TIFF",
  plot = Figure2C,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 16 ,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)


```


```{r Figure2D. Enriched pathways }


pathway_ILRUNgenesUninf24hr <- read.csv("data/ILRUNgenes_Uninf_24hrs_david_default_output.csv") %>% 
  as_tibble() %>%
  arrange(PValue) %>%
  filter(Category == "BIOCARTA" | Category == "KEGG_PATHWAY") %>% 
  filter(PValue < 0.05) %>% 
  separate(Term, c("code", "Pathway"), sep = ":")

gene_ontology_ILRUNgenesUninf24hr <- read.csv("data/ILRUNgenes_Uninf_24hrs_david_default_output.csv") %>% 
  as_tibble() %>%
  arrange(PValue) %>%
  filter(Category == "GOTERM_CC_DIRECT"| Category == "GOTERM_BP_DIRECT") %>% 
  filter(PValue < 0.05) %>% 
  separate(Term, c("code", "Pathway"), sep = ":")

write.csv(pathway_ILRUNgenesUninf24hr , "results/pathway_annotation_ILRUNgenesUninf24hr.csv")

###### Pathway enrichment  ##############
Figure2D <- ggplot(pathway_ILRUNgenesUninf24hr) + 
  geom_bar(aes(x = reorder(Pathway, -PValue), y = Fold_enrichment, fill = PValue), stat = 'identity') +
  coord_flip() +
  ylim(0,12)+
  scale_fill_viridis_c(direction = -1) +
  theme_set(theme_classic(base_size = 6)) +
  theme(legend.key.size = unit(0.3, "cm")) +
  labs(y = "Fold enrichment",
       x = "KEGG or BIOCARTA Pathway",
       fill = "PValue")
  
  
ggsave(
  filename = "results/Figure3A.TIFF",
  plot = Figure2D,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 9 ,
  height = 8,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)



```

```{r Figure3B. Enriched GO terms }


gene_ontology_ILRUNgenesUninf24hr <- read.csv("data/ILRUNgenes_Uninf_24hrs_david_default_output.csv") %>% 
  as_tibble() %>%
  arrange(PValue) %>%
  filter(Category == "GOTERM_CC_DIRECT"| Category == "GOTERM_BP_DIRECT") %>% 
  filter(PValue < 0.05) %>% 
  separate(Term, c("code", "Pathway"), sep = ":") %>%
  mutate(GOterm = str_remove(Pathway, "^.*\\~")) %>% 
  filter(GOterm != "negative regulation of extrinsic apoptotic signaling pathway via death domain receptors")

  

write.csv(gene_ontology_ILRUNgenesUninf24hr , "results/gene_ontology_annotation_ILRUNgenesUninf24hr.csv")

###### Pathway enrichment  ##############
Figure3B <- ggplot(gene_ontology_ILRUNgenesUninf24hr) + 
  geom_bar(aes(x = reorder(GOterm, -PValue), y = Fold_enrichment, fill = PValue), stat = 'identity') +
  coord_flip() +
  scale_fill_viridis_c(direction = -1) +
  theme_set(theme_classic(base_size = 6)) +
  theme(legend.key.size = unit(0.3, "cm")) +
  labs(y = "Fold enrichment",
       x = "GO Term",
       fill = "PValue", 
       tag = "B")
  
  
ggsave(
  filename = "results/Figure3B.TIFF",
  plot = Figure3B,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 9,
  height = 8,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)





```


```{r Figure 3A viral read counts}

cpm_infected <- read.csv("results/coverview_ilrun_SARSCOV2_infected_cpm.csv") %>% 
  mutate(timepoint = str_remove(timepoint, "hr"))
  

cpm_infected$timepoint <- factor(cpm_infected$timepoint,levels =c("6", "24"))
cpm_infected$siRNA <- factor(cpm_infected$siRNA,levels =c("NEG", "ILRUN"))

Figure4A <- ggplot(cpm_infected, aes(x = timepoint, y = cpm)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_y_log10() +
  ylim(0, 2500)+
  theme_set(theme_classic(base_size = 8)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  theme(legend.key.size = unit(0.3, "cm")) +
  labs(tag = "A",
       x= "post-infection (h)",
       y = "Viral RNA (counts per million)")


ggsave(
  filename = "results/Figure4A.TIFF",
  plot = Figure4A,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 10,
  height = 5.5,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)

```


```{r Figure 3B TCID50 vaidation of ILRUN proviral effect}



ILRUN_titres <- read.csv("data/siILRUN_SARSCOV2.csv") %>%
    mutate(treatment = factor(treatment, levels = c("siNEG", "siILRUN", "pCAGGS-GFP", "pCAGGs-ILRUN-FLAG")))


###Summarise for plotting bar graph #####
ILRUN_titres_stats <- ILRUN_titres %>% 
  group_by(treatment) %>% 
  summarise(mean = mean(TCID50), sd = sd(TCID50)) %>% 
  mutate(upper = mean + sd) %>% 
  mutate(lower = mean - sd) %>%
  mutate(treatment = factor(treatment, levels = c("siNEG", "siILRUN", "pCAGGS-GFP", "pCAGGs-ILRUN-FLAG")))

##### plot the data ################
Figure3B <- ggplot(ILRUN_titres, aes(y= TCID50, x= treatment))+
  geom_boxplot(outlier.shape = NA)+
  geom_jitter(width = 0.1)+
  scale_y_log10()+
  theme_set(theme_classic(base_size = 8)) +
  labs(x = "",
       y = expression(paste(log[10],TCID[50], mL^-1)),
       tag = "B")
  
ggsave(
  filename = "results/Figure3B.TIFF",
  plot = Figure3B,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 11,
  height = 5.5,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)



```

```{r Figure 4? PCA for infection}

infection_PCA <- read.csv("results/PCA_data_CoV2genesILRUN24hr_singleplate.csv")


Figure3C <- ggplot(infection_PCA, aes(x = PC1, y = PC2)) +
  geom_point(aes(fill = Condition),color = "black", pch = 21, size = 4)+
  scale_fill_manual(values = c("#E7B800", "#FC4E07"))+
  theme_set(theme_classic(base_size = 10)) +
  theme(legend.position = "none")+
  labs(tag = "C")

ggsave(
  filename = "results/Figure3C.TIFF",
  plot = Figure3C,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 6,
  height = 6,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)




```

```{r Figure 4C Known hots genes for SARSCOV2}


ACE2 <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_expression.csv") %>%
  filter(gene == "ACE2")%>%
  filter(timepoint == "24hr") %>% 
  mutate(siRNA = factor(siRNA, levels = c("NEG", "ILRUN"))) %>% 
  mutate(Infection = factor(Infection, levels = c("Uninfected", "Infected"))) %>% 
  ggplot(aes(x= Infection, y = expression)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  labs(x = "",
       y = "normalized counts",
       title = "ACE2")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))
  

TMPRSS2 <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_expression.csv") %>%
  filter(gene == "TMPRSS2")%>%
  filter(timepoint == "24hr") %>%
  mutate(siRNA = factor(siRNA, levels = c("NEG", "ILRUN"))) %>%
  mutate(Infection = factor(Infection, levels = c("Uninfected", "Infected"))) %>% 
  ggplot(aes(x= Infection, y = expression)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  labs(x = "",
       y = "normalized counts",
       title ="TMPRSS2")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))

FURIN <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_expression.csv") %>%
  filter(gene == "FURIN")%>%
  filter(timepoint == "24hr") %>% 
  mutate(siRNA = factor(siRNA, levels = c("NEG", "ILRUN"))) %>%
  mutate(Infection = factor(Infection, levels = c("Uninfected", "Infected"))) %>% 
  ggplot(aes(x= Infection, y = expression)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  labs(x = "",
       y = "normalized counts",
       title ="FURIN")+
  theme_set(theme_classic(base_size = 8))

CTSL <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_expression.csv") %>%
  filter(gene == "CTSL")%>%
  filter(timepoint == "24hr") %>% 
  mutate(siRNA = factor(siRNA, levels = c("NEG", "ILRUN"))) %>%
  mutate(Infection = factor(Infection, levels = c("Uninfected", "Infected"))) %>% 
  ggplot(aes(x= Infection, y = expression)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  labs(x = "",
       y = "normalized counts",
       title ="CTSL")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))

CTSB <- read.csv("results/DESeq2_ilrun_siRNA_LT07_rev_expression.csv") %>%
  filter(gene == "CTSB")%>%
  filter(timepoint == "24hr") %>%
  mutate(siRNA = factor(siRNA, levels = c("NEG", "ILRUN"))) %>%
  mutate(Infection = factor(Infection, levels = c("Uninfected", "Infected"))) %>% 
  ggplot(aes(x= Infection, y = expression)) +
  geom_boxplot(aes(fill = siRNA)) +
  scale_fill_manual(values = c("#F0E442", "#D55E00"))+
  labs(x = "",
       y = "normalized counts",
       title ="CTSB")+
  theme_set(theme_classic(base_size = 8))


legend <- get_legend(ACE2) 

plot_4C <- plot_grid(ACE2 + theme(legend.position="none"),
                  TMPRSS2 + theme(legend.position="none"),
                  CTSL + theme(legend.position="none"), nrow = 1)

Figure4C <- plot_grid(plot_4C, legend, rel_widths = c(3, .4))

ggsave(
  filename = "results/Figure4C.TIFF",
  plot = Figure4C,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 12 ,
  height = 5.5,
  units = "cm",
  dpi = 300,
  limitsize = TRUE,)
```


```{r  Figure4F ILRUN and ACE2 expression is anti-correllated in TH2-stimulated primary bronchial epithelial cells}

Th2_cytokines <- read.csv("results/GSE113185_Th2.csv")


ILRUN_NHBE_data <- Th2_cytokines %>% 
  filter(transcript == "ilrun") %>%
  mutate(stimulation = factor(stimulation, levels = c("control", "IL-4 + IL-13")))

ACE2_NHBE_data <- Th2_cytokines %>% 
  filter(transcript == "ACE2") %>%
  mutate(stimulation = factor(stimulation, levels = c("control", "IL-4 + IL-13")))

AGT_NHBE_data <- Th2_cytokines %>% 
  filter(transcript == "AGT") %>%
  mutate(stimulation = factor(stimulation, levels = c("control", "IL-4 + IL-13")))



ILRUN_NHBE <- ggplot(ILRUN_NHBE_data, aes( x= stimulation, y = expn)) +
  geom_boxplot(aes(fill = stimulation)) +
  scale_fill_manual(values = c("#999999", "#009E73"))+
  labs(title = "ILRUN", 
       x = "",
       y = "normalized counts")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))


ACE2_NHBE <- ggplot(ACE2_NHBE_data, aes( x= stimulation, y = expn)) +
  geom_boxplot(aes(fill = stimulation)) +
  scale_fill_manual(values = c("#999999", "#009E73"))+
  labs(title = "ACE2", 
       x = "",
       y = "normalized counts")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))

AGT_NHBE <- ggplot(AGT_NHBE_data, aes( x= stimulation, y = expn)) +
  geom_boxplot(aes(fill = stimulation)) +
  scale_fill_manual(values = c("#999999", "#009E73"))+
  labs(title = "AGT", 
       x = "",
       y = "normalized counts")+
  theme_set(theme_classic(base_size = 8))+
  theme(legend.key.size = unit(0.2, "cm"),
        plot.title = element_text(hjust = 0.5, face = "italic"))


legend_NHBE <- get_legend(ILRUN_NHBE) 

plot_4F <- plot_grid(ILRUN_NHBE + theme(legend.position="none"),
                  ACE2_NHBE + theme(legend.position="none"),
                  AGT_NHBE + theme(legend.position="none"), nrow = 1)


Figure4F <- plot_grid(plot_4F, legend_NHBE, rel_widths = c(3, .6))

ggsave(
  filename = "results/Figure4F.TIFF",
  plot = Figure4F,
  device = NULL,
  path = NULL,
  scale = 1,
  width = 12,
  height = 5.5,
  units = "cm",
  dpi = 300,
  limitsize = TRUE)



```

```{r Supplementary Table 1}


ILRUNgenesUninf_24hrs <- read_csv("results/sigILRUNgenesUninf24hr_LT07_rev.csv") %>% 
  as_tibble() %>%
  separate(X1, c("GENE_ID", "X1")) %>% 
  select(-X1) %>% 
  arrange(padj)
write.csv(ILRUNgenesUninf_24hrs, "results/Supp1_ILRUNgenesUninf24hrs.csv")

ILRUNgenesUninf_6hrs <- read_csv("results/sigILRUNgenesUninf6hr_LT07_rev.csv")  %>% 
  as_tibble() %>%
  separate(X1, c("GENE_ID", "X1")) %>% 
  select(-X1) %>% 
  arrange(padj)
write.csv(ILRUNgenesUninf_6hrs, "results/Supp1_ILRUNgenesUninf6hrs.csv")

ILRUNgenesInf_24hrs <- read_csv("results/sigILRUNgenesInf24hr_LT07_rev.csv") %>% 
  as_tibble() %>%
  separate(X1, c("GENE_ID", "X1")) %>% 
  select(-X1) %>% 
  arrange(padj)
write.csv(ILRUNgenesInf_24hrs, "results/Supp1_ILRUNgenesInf24hrs.csv")

ILRUNgenesInf_6hrs <- read_csv("results/sigILRUNgenesInf6hr_LT07_rev.csv")  %>% 
  as_tibble() %>%
  separate(X1, c("GENE_ID", "X1")) %>% 
  select(-X1) %>% 
  arrange(padj)
write.csv(ILRUNgenesInf_6hrs, "results/Supp1_ILRUNgenesInf6hrs.csv")

```

